File: backend/src/main/java/com/vet/system/model/MedicalHistory.java
package com.vet.system.model;

import com.fasterxml.jackson.annotation.JsonBackReference;
import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import java.time.LocalDate;

@Entity
@Table(name = "medical_histories")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class MedicalHistory {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    private String condition;
    private String treatment;
    private LocalDate date;
    private String notes;
    
    @ManyToOne
    @JoinColumn(name = "pet_id")
    private Pet pet;
    
    @JsonBackReference
    @ManyToOne
    @JoinColumn(name = "appointment_id")
    private Appointment appointment;
} 

File: frontend/src/app/models/medical-history.ts
export interface MedicalHistory {
  id?: number;
  condition: string;
  treatment: string;
  date: string;  // ISO date string
  notes: string;
  pet: {
    id: number;
    name: string;
    breed: string;
  };
  appointment?: {
    id: number;
    dateTime: string;
  };
}

export interface NewMedicalHistory extends Omit<MedicalHistory, 'id'> {} 